cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/service/http/HttpHandlerGeneratorTest.java

```diff
@@ -261,33 +309,119 @@ protected FileUploadHandler createHandler() {
       HttpURLConnection urlConn = (HttpURLConnection) new URL(
         String.format("http://%s:%d/content/upload/test.txt",
                       bindAddress.getHostName(), bindAddress.getPort())).openConnection();
+      try {
+        urlConn.setReadTimeout(2000);
+        urlConn.setDoOutput(true);
+        urlConn.setRequestMethod("PUT");
+        // Set to use default chunk size
+        urlConn.setChunkedStreamingMode(-1);
+
+        // Write over 150MB of data
+        // One fragment is 10K of data
+        byte[] fragment = Strings.repeat("0123456789", 1024).getBytes(Charsets.UTF_8);
+        File localFile = TEMP_FOLDER.newFile();
+        try (
+          OutputStream os = urlConn.getOutputStream();
+          FileOutputStream fos = new FileOutputStream(localFile)
+        ) {
+          for (int i = 0; i < 20000; i++) {
+            os.write(fragment);
+            fos.write(fragment);
+          }
+        }
 
-      urlConn.setReadTimeout(2000);
-      urlConn.setDoOutput(true);
-      urlConn.setRequestMethod("PUT");
-      // Set to use default chunk size
-      urlConn.setChunkedStreamingMode(-1);
-
-      // Write over 150MB of data
-      // One fragment is 10K of data
-      byte[] fragment = Strings.repeat("0123456789", 1024).getBytes(Charsets.UTF_8);
-      File localFile = TEMP_FOLDER.newFile();
-      try (
-        OutputStream os = urlConn.getOutputStream();
-        FileOutputStream fos = new FileOutputStream(localFile)
-      ) {
-        for (int i = 0; i < 20000; i++) {
-          os.write(fragment);
-          fos.write(fragment);
+        Assert.assertEquals(200, urlConn.getResponseCode());
+
+        // File written on the remote end should be > 150MB and should be the same as the local one
+        File remoteFile = new File(outputDir, "test.txt");
+        Assert.assertTrue(remoteFile.length() > 150 * 1024 * 1024);
+        Assert.assertEquals(Files.hash(localFile, Hashing.md5()), Files.hash(remoteFile, Hashing.md5()));
+      } finally {
+        urlConn.disconnect();
+      }
+    } finally {
+      service.stopAndWait();
+    }
+  }
+
+  @Test
+  public void testContentProducer() throws Exception {
+    MetricsContext noOpsMetricsContext =
+      new NoOpMetricsCollectionService().getContext(new HashMap<String, String>());
+    HttpHandlerFactory factory = new HttpHandlerFactory("/content", noOpsMetricsContext);
+
+    // Create the file upload handler and starts a netty server with it
+    final File outputDir = TEMP_FOLDER.newFolder();
+    HttpHandler httpHandler = factory.createHttpHandler(
+      TypeToken.of(FileHandler.class), new AbstractDelegatorContext<FileHandler>() {
+        @Override
+        protected FileHandler createHandler() {
+          return new FileHandler(outputDir);
         }
+      });
+
+    NettyHttpService service = NettyHttpService.builder()
```

Comment:
so you are testing this directly against a netty-http. Why not against a CDAP service?